### PR TITLE
Feature/RIFEX-43/Active-domain-search

### PR DESCRIPTION
**Pull request by rorolopetegui**
This PR implements the functionality to get all details for a domain searched in the search-bar component.
-Resolver combo select implemented (Multicrypto)
-Add new chain address implmemented

SearchDomainActive

### CODE REVIEW

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/resolver/index.js]: This method returns the remaining days if there is no error, so why not name it accordly?

File: ui/app/rif/actions.js
Comment (marcosmartinez7) [inline ui/app/rif/actions.js]: This should resolve to `details ` not `domainDescription ` right? That variable is not defined.

File: ui/app/rif/components/searchDomains.js
Comment (marcosmartinez7) [inline ui/app/rif/components/searchDomains.js]: I think `domainToRegister` isnt clear. That variable name tells that we are trying to register a domain, but the generic case its just a search. Depending on the results the app goes to the registration or details. 

So i would name it just `domain` or something

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/resolver/index.js]: Oh! There in last commit i just renamed all to let the things more understandable, that was just a copy pasted function from the manager haha

File: ui/app/rif/actions.js
Comment (rorolopetegui) [reply ui/app/rif/actions.js]: You right, as i was just testing the plugin with the background console i didn't give importance to the UI. In last commit its fixed btw

File: ui/app/rif/components/searchDomains.js
Comment (rorolopetegui) [reply ui/app/rif/components/searchDomains.js]: You right, i renamed this vars!

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/resolver/index.js]: I think it would be nice to refactor this, in order to use `async/await`, just to avoid promise `.then `cascade

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/resolver/index.js]: We need to make a standard on contract invokations.

Take a look at `rns-delegate ` `call ` method https://github.com/anarancio/nifty-wallet/pull/8/files#diff-fd3f70ef0d29a042399ae36fc5301309

Lets discuss this with @JONAF2103 at tomorrow daily

File: package.json
Comment (marcosmartinez7) [inline package.json]: Is this really necessary? We already have web3-utils for that right?

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [reply app/scripts/controllers/rif/rns/resolver/index.js]: Also, i dont like the idea of invoking contracts from the controller. I find the contorller as a top level component. For that we have the service layer. 

Lets discuss that too

File: ui/app/rif/actions.js
Comment (marcosmartinez7) [inline ui/app/rif/actions.js]: Can we use console.debug for those cases?

File: ui/app/rif/components/searchDomains.js
Comment (marcosmartinez7) [inline ui/app/rif/components/searchDomains.js]: Same here, lets use console.debug

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [reply app/scripts/controllers/rif/rns/resolver/index.js:85]: marcosmartinez7 17 hours ago Collaborator
We need to make a standard on contract invokations.

Take a look at rns-delegate call method https://github.com/anarancio/nifty-wallet/pull/8/files#diff-fd3f70ef0d29a042399ae36fc5301309

Lets discuss this with @JONAF2103 at tomorrow daily

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (JONAF2103) [reply app/scripts/controllers/rif/rns/resolver/index.js]: If he do that it can be a little problematic, since if you use await you will need to put your function as async always. 
I think this is a matter of taste, it depends on the situation, maybe you run a method that needs to do something on the callback but you don't want to wait for it and with async/await you can't do that because you are waiting for the response.
I know that in this case is a lot so maybe we should change all the delegate expose api methods to async instead, and then start thinking about when is good to use async/await and when is not, probably we will need to keep a balance on that.

File: package.json
Comment (JONAF2103) [reply package.json]: Yes but i was added after this, we should change this anyway.

File: rif.config.js
Comment (JONAF2103) [inline rif.config.js]: Remove this, remember this file should be clean, anyway i know that the .gitignore is failing on skip this file, we should take a look at that too.

**Review by JONAF2103 — CHANGES_REQUESTED** [review]
In general looks good but some changes are needed, also be careful with the linter, for some reason is not right, maybe you are using visual studio, please check that, thanks! :)

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [reply app/scripts/controllers/rif/rns/resolver/index.js]: I dont have in mind any situation where async/await must not be used, but sure we can discuss that.

Take into account that when the callback executes is after the principal operation finish, so is the same as wait for the result with `await`

File: rif.config.js
Comment (marcosmartinez7) [reply rif.config.js]: Maybe this config can be part of a `.env ` file to make it more clear for a dev?

Or move this to another config file that is already used by nifty?

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (JONAF2103) [reply app/scripts/controllers/rif/rns/resolver/index.js]: Why you should force to use async to the caller always? that's what i don't understand, i mean if you use async you are forcing your caller to use await if he wants the result, instead with promises you can await or just manage the response with then, is matter of taste, if not, promises shouldn't exists and no one should use them, i think is just different schools that's all.

Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/resolver/index.js]: Hey guys, the code has change a little, now it's using promises.all for this.

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/resolver/index.js]: It have changed, check the new code please. Its implemented with .call from @JONAF2103

File: package.json
Comment (rorolopetegui) [reply package.json]: Deleted now, and the code that was using this, now is using the web3-utils

File: ui/app/rif/actions.js
Comment (rorolopetegui) [reply ui/app/rif/actions.js]: Changed all console.log to .debug, thnks

File: ui/app/rif/components/searchDomains.js
Comment (rorolopetegui) [reply ui/app/rif/components/searchDomains.js]: Changed all console.log to .debug, thnks

File: rif.config.js
Comment (rorolopetegui) [reply rif.config.js]: When the PR is ready i'll delete this, i take note of this for now, so i dont have to be carefully looking to no upload this.

File: ui/app/rif/constants/index.js
Comment (marcosmartinez7) [inline ui/app/rif/constants/index.js]: Why is this name discrimination needed?

File: app/scripts/controllers/rif/rns/classes/resolverNetwork.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/classes/resolverNetwork.js]: Fix this boilerplate comments

File: app/scripts/controllers/rif/rns/classes/resolverNetwork.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/classes/resolverNetwork.js]: What is the purpose of this class? ResolverNetwork doesnt say anything to me

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/resolver/index.js]: lets call this `chainAddrChangedEvent ` instead of `myEvent`

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/resolver/index.js]: `ResolverNetwork `is more like a tuple of `(chainId,address)`, i would call it `ChainAddress`. I think is more representative.

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/resolver/index.js]: `network` doesnt seems a good variable name to me. Here you have the event that youre fetching, not a network.

What about `chainAddrChangedEvent`? or just `event`?

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/resolver/index.js]: I dont think this is ok. 

If we have, for example, these events: 

0, 0x1
2, 0x1
3, 0x2
3, 0x4

This will search for the second occurrence of the chain id 3 (3, 0x4), and overrides the previous (3, 0x2) entry. And thats not ok, because those are two different addresses. 

We need to get the latest event for the same address and the same chainId.

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/resolver/index.js]: I dont find the function name nor the documentation about this method related with its behaviour. 

This invokes `setChainAddr`, and that set an address into the chain map of the domain node.
```

 function setChainAddr (bytes32 node, bytes4 chain, string memory addrValue) public onlyOwner(node) {
        chainAddresses[node][chain].addr = addrValue;
        if (chain == RSK_CHAIN_ID) {
            address _addr = addressHelper.stringToAddress(addrValue);
            emit AddrChanged(node, _addr);
        } else {
            emit ChainAddrChanged(node, chain, addrValue);
        }
    }
```

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/resolver/index.js]: Why `domainNameResolver`? isnt that parameter the bytes32 node of the domain? Its just the `domainName`, is not related with the resolver at all

**Review by marcosmartinez7 — CHANGES_REQUESTED** [review]

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/resolver/index.js]: This works fine, because it's filtering by domain name in this line: 
`chainAddrChangedEvent = this.multiChainresolverContractInstance.ChainAddrChanged({node: namehash.hash(domainNameResolver)}, {fromBlock: 0, toBlock: 'latest'});`
So it will be looking for events for a particular domainName. Iterating in the forEach, its replacing the older objects in the array to return.

Comment (marcosmartinez7) [reply app/scripts/controllers/rif/rns/resolver/index.js]: Thats right!

File: app/scripts/controllers/rif/rns/classes/resolverNetwork.js
Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/classes/resolverNetwork.js]: This was deleted in previous commit.

File: app/scripts/controllers/rif/rns/classes/resolverNetwork.js
Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/classes/resolverNetwork.js]: Thanks, it was renamed

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/resolver/index.js]: Yep, you right, i change the name to that

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/resolver/index.js]: Changed to that name!

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/resolver/index.js]: As i use it just for iterating, i changed it to event

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/resolver/index.js]: Changed in previous commit!

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/resolver/index.js]: Changed in previous commit!

File: rif.config.js
Comment (marcosmartinez7) [inline rif.config.js]: I think that this file should have all the mainnet addresses.

And if the user changes the network to RSK Testnet then we need to change the addresses.

Maybe the object should have mainnet and testnet keys.

Btw, this is not strictly related with this pr, but what happens if a user, connected to ethereu, try to use the "my domains" feature?

File: ui/app/rif/components/modal/addNewTokenNetworkAddress/addNewChainAddressToResolver.js
Comment (JONAF2103) [inline ui/app/rif/components/modal/addNewTokenNetworkAddress/addNewChainAddressToResolver.js]: I don't get it, is this deleted or not? down below says that you deleted this file, anyway i think this file is not needed, i mean you should use the generic modal instead of adding new modal templates, that's why we have the elements logic and everything to use that. Can you explain why you have this?

File: ui/app/rif/constants/index.js
Comment (JONAF2103) [inline ui/app/rif/constants/index.js:1]: I recommend you to put constants inside objects to diferentiate what constants are from where, is just to organize better the constants inside this file.

**Review by JONAF2103 — CHANGES_REQUESTED** [review]

File: rif.config.js
Comment (JONAF2103) [reply rif.config.js]: We have to control that, probably or hide the menu or discuss about if we are going to provide support for multiple chains outside the RSK ecosystem.

File: rif.config.js
Comment (JONAF2103) [reply rif.config.js]: @marcosmartinez7 the .env can't be done because it doesn't work properly with extensions, nifty doesn't have any config file, it has js files with config all over the place, we should discuss how to solve this properly.

File: rif.config.js
Comment (marcosmartinez7) [reply rif.config.js]: Yes, also, there is no sense in all the rif operations for other chains.

File: ui/app/rif/components/modal/addNewTokenNetworkAddress/addNewChainAddressToResolver.js
Comment (rorolopetegui) [reply ui/app/rif/components/modal/addNewTokenNetworkAddress/addNewChainAddressToResolver.js]: This was moved to the folder of Domain Active in next commit.

File: ui/app/rif/constants/index.js
Comment (rorolopetegui) [reply ui/app/rif/constants/index.js:1]: This was moved in next commit, so as you said, we have sepparated logic here.

File: rif.config.js
Comment (rorolopetegui) [reply rif.config.js]: I created a ticket to differentiate between Mainnet and Testnet, also to not show Rif Tools when others Non-Rsk chains are selected

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/resolver/index.js]: What about this TODO? The UI logic includes the case when there is no resolver set?

**Review by marcosmartinez7 — APPROVED** [review]

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (JONAF2103) [inline app/scripts/controllers/rif/rns/resolver/index.js]: You will need to start using the listener when we merge the subdomain list PR.

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (JONAF2103) [inline app/scripts/controllers/rif/rns/resolver/index.js]: Same thing here about the listener if you want to control the action after the transaction.

**Review by JONAF2103 — APPROVED** [review]
LGTM

File: app/scripts/controllers/rif/rns/resolver/index.js
Comment (rorolopetegui) [reply app/scripts/controllers/rif/rns/resolver/index.js]: Yep, i forget to delete this!

**Pull request merged**
